packages/core/src/__clients__/core.ts

```diff
@@ -0,0 +1,3480 @@
+
+    const providers = walletProviders ?? (await this.getWalletProviders());
+    const matching = providers.filter(
+      (p) =>
+        p.info?.name?.toLowerCase().replace(/\s+/g, "-") === wallet.walletId &&
+        p.connectedAddresses.length > 0,
+    );
+
+    for (const provider of matching) {
+      const timestamp = toExternalTimestamp();
+
+      for (const address of provider.connectedAddresses) {
+        const account: ConnectedWalletAccount = {
+          walletAccountId: `${wallet.walletId}-${provider.interfaceType}-${address}`,
+          organizationId: session.organizationId,
+          walletId: wallet.walletId,
+          curve: isEthereumWallet(provider) ? Curve.SECP256K1 : Curve.ED25519,
+          pathFormat: "PATH_FORMAT_BIP32",
+          path: WalletSource.Connected,
+          source: WalletSource.Connected,
+          addressFormat: isEthereumWallet(provider)
+            ? "ADDRESS_FORMAT_ETHEREUM"
+            : "ADDRESS_FORMAT_SOLANA",
+          address,
+          createdAt: timestamp,
+          updatedAt: timestamp,
+          ...getWalletAccountMethods(
+            this.walletManager!.connector!.sign.bind(
+              this.walletManager!.connector,
+            ),
+            provider,
+          ),
+          ...(isSolanaWallet(provider) && { publicKey: address }),
+        };
+
+        connected.push(account);
+      }
+    }
+
+    return connected;
+  };
+
+  /**
+   * Signs a message using the specified wallet account.
+   *
+   * - Supports both embedded and connected wallets.
+   * - For **connected wallets**:
+   *   - Delegates signing to the wallet provider’s native signing method.
+   *   - **Important:** For Ethereum wallets (e.g., MetaMask), signatures follow [EIP-191](https://eips.ethereum.org/EIPS/eip-191).
+   *     The message is automatically prefixed with `"\x19Ethereum Signed Message:\n" + message length`
+   *     before signing. As a result, this signature **cannot be used as a raw transaction signature**
+   *     or broadcast on-chain.
+   * - For **embedded wallets**, uses the Turnkey API to sign the message directly.
+   * - Automatically handles message encoding and hashing based on the wallet account’s address format,
+   *   unless explicitly overridden.
+   *
+   * @param params.message - message to sign.
+   * @param params.walletAccount - wallet account to use for signing.
+   * @param params.encoding - override for the payload encoding (defaults to the encoding appropriate for the address type).
+   * @param params.hashFunction - override for the hash function (defaults to the hash function appropriate for the address type).
+   * @param params.stampWith - stamper to tag the signing request (e.g., Passkey, ApiKey, or Wallet).
+   * @param params.addEthereumPrefix - whether to prefix the message with Ethereum's `"\x19Ethereum Signed Message:\n"` string.
+   *   - If `true` (default for Ethereum), the message is prefixed before signing.
```

Comment:
@moeodeh3 This says that we default to true for ethereum but I don't see below in the function where we are setting this value to true in the event that the address format is ethereum